DiscordBot/Commands/EventCommands.cs

```diff
@@ -217,6 +219,72 @@ public async Task EditEvent(CommandContext context, InteractivityModule interact
             await EditEventField(context, interactivity, eventKey.Value, editField, eventEmbed);
         }
 
+        private static async Task CreateSignupSheet(CommandContext context, InteractivityModule interactivity)
+        {
+            await context.RespondAsync($"{context.Member.Mention} - what is the event key? (use the ``list`` option to find out)");
+            var eventKey = await GetUserIntResponse(context, interactivity);
+            if (eventKey == null)
+            {
+                return;
+            }
+
+            var eventEmbed = await GetEventEmbed(context, eventKey.Value);
+            if (eventEmbed == null)
+            {
+                return;
+            }
+
+            await context.RespondAsync($"{context.Member.Mention} - start signups for this event? - (``yes``/``no``)", embed: eventEmbed);
+            var confirmationResponse = await GetUserConfirmation(context, interactivity);
+            if (confirmationResponse == null || confirmationResponse == false)
+            {
+                return;
+            }
+
+            await SendSignupMessage(context, eventKey.Value);
+        }
+
+        private static async Task SendSignupMessage(CommandContext context, int eventKey)
```

Comment:
This method is pretty long. Can you split it up a bit? 🙂

Reply:
I assume you're referring to `SendSignupMessage()`?

Reply:
I'm having some troubles doing this, can I have some suggestions?

Reply:
So I think splitting out creating the sign up embed would be useful. Depending how big that split out method turns out to be, you could even split that up further so that you have separate methods for adding participants to it, adding responses to it and adding response options. Does that make sense? 🙂

Reply:
I'm finding it hard to do that because adding the reactions (which needs to be done after the message is sent) requires information from the optionsList variable (which is also used for the embed). Should I include that as an output parameter as well as returning the embed?

Reply:
In that case I would get the userResponseDictionary and optionsList here in this method. Then you can pass them in to the method to create the embed. I would not recommend using an output parameter as code using an output parameter can be a bit confusing 🙂